### PR TITLE
Fix trivial problems reported by SpotBugs

### PR DESCRIPTION
**Pull request by danielmitre**
Fixed problems reported by SpotBugs that don't change signatures and seems to change no functionality.

+ [x] Upgrade `java.text.DateFormat` to `java.time.*` in possible multi-threaded code.
+ [x] Fix `Serializable` class with non-`Serializable` members
+ [x] Fix `instanceof` comparing to wrong class
+ [x] Enhance performance with String manipulation
+ [x] Fix resources not closed
+ [x] Fix objects with custom `equals` implementation relying on default `hashcode`

The report was generated with the command `mvn com.github.spotbugs:spotbugs-maven-plugin:4.2.3:check`

### CODE REVIEW

File: src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java [hunk 1]
```diff
@@ -18,17 +18,20 @@
...
   public void createToken() throws Exception {
     Properties properties = new Properties();
-    FileInputStream input = new FileInputStream(CONFIG_FILE_PATH);
-    properties.load(input);
 
-    String url = properties.getProperty(SapsPropertiesConstants.Openstack.IdentityService.API_URL);
-    String projectId = properties.getProperty(SapsPropertiesConstants.Openstack.PROJECT_ID);
-    String userId = properties.getProperty(SapsPropertiesConstants.Openstack.USER_ID);
-    String password = properties.getProperty(SapsPropertiesConstants.Openstack.USER_PASSWORD);
+    try (FileInputStream input = new FileInputStream(CONFIG_FILE_PATH)) {
```
Comment (thiagomanel) [inline src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java:22]: are we missing a catch here?

File: src/test/java/saps/common/core/storage/swift/SwiftAPIClientTest.java [hunk 2]
```diff
@@ -16,9 +16,10 @@
 
   private Properties loadConfigFile(String path) throws IOException {
     Properties properties = new Properties();
-    FileInputStream input = new FileInputStream(path);
-    properties.load(input);
-    return properties;
+    try (FileInputStream input = new FileInputStream(path)) {
```
Comment (thiagomanel) [inline src/test/java/saps/common/core/storage/swift/SwiftAPIClientTest.java:19]: same here. I didn't get the point of a catch-less try

**Review by thiagomanel — COMMENTED** [review]
I didn't understand why are we adding a try clause without a catch. why?

File: src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java (hunk 1 above)
Comment (danielmitre) [reply src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java:22]: This is a try-with-resources block, is a syntax sugar to auto-close the resource (the FileInputStream in this case).

File: src/test/java/saps/common/core/storage/swift/SwiftAPIClientTest.java (hunk 2 above)
Comment (danielmitre) [reply src/test/java/saps/common/core/storage/swift/SwiftAPIClientTest.java:19]: Same: the purpose is to close ensure the FileInputStream will be closed outside this scope.

File: src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java (hunk 1 above)
Comment (thiagomanel) [reply src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java:22]: does it depend on an specific jdk version onward?

Comment (danielmitre) [reply src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java:22]: It is supported since jdk 7 (2011)

Comment (thiagomanel) [reply src/test/java/saps/common/core/storage/swift/KeystoneV3IdentityPluginTest.java:22]: yep, I'm old school, I guess. Nice!

**Pull request merged**